toolkit/docs/how_it_works/1_initial_prep.md

```diff
@@ -69,10 +72,10 @@ The `depsearch` tool is used to list all packages which depend on another set of
 
 #### grapher
 The `grapher` tool is responsible for creating the initial dependency graph from the parsed spec files (see [Dependency Graphing](3_package_building.md#dependency-graphing)). It outputs a graph based on all local packages and their dependencies. It makes no attempt to optimize the graph or find unresolved dependencies.
-#### graphoptimizer
-The `graphoptimizer` tool takes the output from the `grapher` tool and looks for existing copies of the local rpms (see [Stage 2: Graphoptimizer](3_package_building.md#stage-2-graphoptimizer)). If it finds one it will mark the node as `up-to-date`. If a package needs to be re-built it will mark any packages which rely on it as needing to be re-built. The `graphoptimizer` tool bases its decisions on the currently selected image configuration.
+#### graphanalytics
+`graphanalytics` is an optional tool that analyzes the built graph from `scheduler` and generates a summary with information regarding any packages that are blocked from building. The summary includes the packages that are most blocking other packages from building and the packages closest to being ready to build.
```

Comment:
>most [](start = 223, length = 4)

Nit: does the summary also include packages, which are "least blocking"?:) What I'm getting at, is that it feels like I might not get a full summary of all blocking packages.

Reply:
There's an optional argument to `graphanalytics` to configure how many entries it will show. If a person wants to see the whole summary they may do so through that variable.

It looks like the flag didn't get plumbed through to the makefile. I'll see about adding it.